storage/remote/storage.go

```diff
@@ -86,22 +88,32 @@ func (s *Storage) ApplyConfig(conf *config.Config) error {
 
 	// Update read clients
 
-	clients := []*Client{}
+	s.queryables = make([]storage.Queryable, 0, len(conf.RemoteReadConfigs))
 	for i, rrConf := range conf.RemoteReadConfigs {
 		c, err := NewClient(i, &ClientConfig{
 			URL:              rrConf.URL,
 			Timeout:          rrConf.RemoteTimeout,
 			HTTPClientConfig: rrConf.HTTPClientConfig,
-			ReadRecent:       rrConf.ReadRecent,
 		})
 		if err != nil {
 			return err
 		}
-		clients = append(clients, c)
-	}
 
-	s.clients = clients
-	s.externalLabels = conf.GlobalConfig.ExternalLabels
+		var q storage.Queryable
+		q = QueryableClient(c)
+		q = ExternablLabelsHandler(q, conf.GlobalConfig.ExternalLabels)
+		if len(rrConf.RequiredMatchers) > 0 {
+			ms := make([]*labels.Matcher, 0, len(rrConf.RequiredMatchers))
+			for k, v := range rrConf.RequiredMatchers {
+				ms[i] = &labels.Matcher{Name: string(k), Value: string(v)}
+			}
```

Comment:
To make this little bit of code a little cleaner, you could hide the conversion from `model.Labels` to `[]*labels.Matcher` in a function.

Reply:
Or in the constructor for `RequiredMatchersFilter`

Reply:
I wouldn't want to use both model.LabelSet and labels.Matcher in the read point. It's just a log easier to use model.LabelSet in the config. I'll extract the code into a utility function.